### PR TITLE
Automated tests and display the test with the result.

### PR DESCRIPTION
**Pull request by te4-gabriel-stark**
#40

### CODE REVIEW

File: public/script.js [hunk 1]
```diff
@@ -122,14 +122,20 @@ function printRepos(repos, githubLinks) {
     }
 }
 
-function printForks(code, userFork, githubLink, i) {
+function printForks(code, userFork, githubLink, i, manifest) {
```
Comment (itggot-linus-styren) [inline public/script.js]: Are the parameters `i` and `manifest` optional as hinted at line 68-70? Have you considered writing some documentation about the printForks function.

File: public/script.js [hunk 2]
```diff
@@ -182,4 +188,17 @@ function noManifest(message, githubLink){
     baseHtml[9] = `                        <a class="fork" href="${githubLink}">Show On GitHub</a>`;
     baseHtml = baseHtml.join("\n");
     output.firstElementChild.innerHTML += `${baseHtml}`;
-}
\ No newline at end of file
+}
+
+function testTest(code, manifest) {
```
Comment (itggot-linus-styren) [inline public/script.js]: The function name "testTest" is accurate but maybe a name such "runTest" would avoid confusion?

File: public/script.js [hunk 3]
```diff
@@ -182,4 +188,17 @@ function noManifest(message, githubLink){
...
+    let results = [];
+    code = code.split("\n");
+    code.shift();
+    if(code[code.length - 1] == "") code.pop();
+    code.pop();
+    let fun = new Function("num1,num2", code.join("\n"));
+    for(let i = 0; i < manifest.tests.length; i++) {
+        manifest.tests[i].expected == fun(...manifest.tests[i].arguments) ? results.push("Passed") : results.push("Failed");
```
Comment (itggot-linus-styren) [inline public/script.js]: What is the difference between calling the function directly and using .bind (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind)?

File: public/script.js [hunk 4]
```diff
@@ -182,4 +188,17 @@ function noManifest(message, githubLink){
...
\ No newline at end of file
+}
+
+function testTest(code, manifest) {
+    let results = [];
+    code = code.split("\n");
+    code.shift();
+    if(code[code.length - 1] == "") code.pop();
+    code.pop();
+    let fun = new Function("num1,num2", code.join("\n"));
```
Comment (itggot-linus-styren) [inline public/script.js]: `const` is normally used when the variable doesn't change.

Comment (itggot-linus-styren) [inline public/script.js]: Avoiding all scenarios of arbitrary code execution would require some extra work, refer to this resource for some interesting insights: https://stackoverflow.com/a/46374395

File: public/script.js [hunk 5]
```diff
@@ -68,7 +68,7 @@ async function getForks(e) {
             if(fileContent[i].includes(`isn't`) == true){
```
Comment (te4-david-petersson) [inline public/script.js]: consider changing changing 'isn't' to a constant, for example MANIFEST_KEYWORD

File: public/script.js (hunk 4 above)
Comment (te4-gabriel-stark) [reply public/script.js]: Eftersom att bind inte funkade så kunde jag inte lösa problemet.

File: public/script.js (hunk 3 above)
Comment (te4-gabriel-stark) [reply public/script.js]: Jag hittade inget som kunde hjälpa mig i detta fallet där det inte riktigt funkade.

File: public/script.js (hunk 5 above)
Comment (te4-gabriel-stark) [reply public/script.js]: Changed it so that it returns null if it there is no code instead of a string.

**Pull request merged**